newton/_src/sim/builder.py

```diff
@@ -483,6 +516,69 @@ def __init__(self, up_axis: AxisType = Axis.Z, gravity: float = -9.81):
         self.equality_constraint_polycoef = []
         self.equality_constraint_key = []
         self.equality_constraint_enabled = []
+        # Custom attributes (user-defined per-frequency arrays)
+        self.custom_attributes: dict[str, CustomAttribute] = {}
+
+    @staticmethod
+    def _default_for_dtype(d: object) -> Any:
```

Comment:
I think this method can also go into the `CustomAttribute` class? Then you could set the default in its `__post_init__` method.